### PR TITLE
Update synapse installation doc

### PR DESCRIPTION
**Pull request by alfogrillo**
### Description
This PR updates the documentation for installing a local synapse instance on Mac OS machines.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -472,20 +472,30 @@ Then, open ``MatrixSDK.xcworkspace``.
...
 
+Before you install synapse you may need few dependencies to be installed on Mac OS:
+
+- **Homebrew**: run ``/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)”``. More information can be found here https://brew.sh
+- **python 3**: downloading the latest stable version should be fine. Download the ``.pkg`` and install it from here https://www.python.org/downloads/
+- **pipx**: with python installed run ``pip3 install --user pipx``
+- **Rust**: run ``curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh``. more information can be found here https://www.rust-lang.org/tools/install
+- **icu4c**: Run ``brew install icu4c``
+- **Update env variables for icu4c**: if you use zsh run ``echo 'export PATH="/opt/homebrew/opt/icu4c/bin:$PATH"' >> ~/.zshrc``. Otherwise try to update ``.bash_profile`` in the same way. If you can’t find ``icu4c`` in the default homebew folder try to run ``which icu4c`` to find the correct path and update the command above.
+- **pg_config**: you can get it by running ``brew install postgresql``
```
Comment (Johennes) [inline README.rst:491]: In https://github.com/matrix-org/synapse/pull/13480/files, we postulated that libpq would be enough. Had you tried that out?

Comment (Johennes) [reply README.rst:491]: (original issue: https://github.com/matrix-org/synapse/issues/13464)

Comment (alfogrillo) [reply README.rst:491]: Do you mean `libpq` instead of `postgresql`?

**Comment by clokep** [comment]
Would it be possible to point to the Synapse docs and ensure those are correct instead of duplicating them here?

File: README.rst (hunk 1 above)
Comment (Johennes) [reply README.rst:491]: Yeah, exactly

Comment (Johennes) [reply README.rst:491]: Hm, ok sorry I may have mixed this up then. It's odd though that postgres is not mentioned in the official install instructions?

https://matrix-org.github.io/synapse/latest/setup/installation.html

**Comment by alfogrillo** [comment]
> Would it be possible to point to the Synapse docs and ensure those are correct instead of duplicating them here?

@clokep for what I saw the [official documentation](https://matrix-org.github.io/synapse/latest/setup/installation.html) explains how to install synapse using prebuilt packages or as a PyPI module.
The goal of this update is being able to run a demo from sources which is useful especially for devs.
Do you think this knowledge is useful outside this repo? If yes, where do you think it should go?

**Comment by clokep** [comment]
> > Would it be possible to point to the Synapse docs and ensure those are correct instead of duplicating them here?
> 
> @clokep for what I saw the [official documentation](https://matrix-org.github.io/synapse/latest/setup/installation.html) explains how to install synapse using prebuilt packages or as a PyPI module. The goal of this update is being able to run a demo from sources which is useful especially for devs. Do you think this knowledge is useful outside this repo? If yes, where do you think it should go?

There's a section on what you need to install for macOS: https://matrix-org.github.io/synapse/latest/setup/installation.html#macos

There's also information in the developer guide of requirements: https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#2-what-do-i-need

That seems to already mention rust though.

**Comment by alfogrillo** [comment]
> There's a section on what you need to install for macOS: https://matrix-org.github.io/synapse/latest/setup/installation.html#macos

This seems to be under PyPI. The scope is probably different.

> There's also information in the developer guide of requirements: https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#2-what-do-i-need
> 
> That seems to already mention rust though.

Yeah it's mentioned. 
So I guess most of dependencies are covered.
I think just **icu4c** and **pg_config** aren't.

The problem I guess is that there aren't platform specific sections for devs?
If so, it seems like there isn't a good spot where to merge this info in.

**Review by stefanceriu — APPROVED** [review]
No matter what you decide these steps worked on my side and look good to me 👍

**Pull request merged**